### ISSUE TITLE
I2I: Add data disclaimer on interactive components

### ISSUE THREAD

**Issue by mszylkowski** [ampproject/amphtml#31447]
![image](https://user-images.githubusercontent.com/22420856/101213417-59f97d00-3648-11eb-9bd2-c6a920013858.png)

For user privacy concerns, we want to give users the url where their response will be sent. We will add a consent dialog with info on what the backend URL is for an interactive component.
A `learn-more-url` attribute will be provided for stories to specify a link to their Terms of Service if they want to provide one. Otherwise the "Learn more" link will not be shown.

```html
<amp-story-interactive-poll
    endpoint="https://backend.googleapis.com/v1/interactives"
    learn-more-url="https://google.com/tos"
    option-1-text="Pizza"
    option-2-text="Burgers"></amp-story-interactive-quiz>
```

We will also have a _lookup table_ for backend owners to specify their attributes so that they are automatically applied backwards to all the stories that have the domain specified. Backend owners will need to submit a PR to be added to this table (or as the WG to do so), and will only be able to claim a backend if they own the domain. Backend urls that don't have an entry in this table will still show the domain ("Your response will be sent to <url>").

```json
{
    "https://backend.googleapis.com/v1/interactives": {
        "entityName": "Google",
        "learnMoreUrl': 'https://google.com/tos"
    },
    "https://amp.dev/api/interactives": {
        "entityName": "AMP",
        "learnMoreUrl": "https://amp.dev/stories/tos"
    }
}
```

https://codepen.io/mszylkowski/details/QWKborj

/cc @ampproject/wg-stories

**Comment by gmajoulet** [comment]
Can you provide a full `<amp-story-interactive-*` example with the attributes available? What is displayed if you only use the attributes, I'm especially curious about the `entity`?

What is the default, if you don't specify any attribute nor send a PR to be in that list?

**Comment by mszylkowski** [comment]
Updated the description to answer those questions.

TL;DR: You cannot display an entity name if your backend is not in the lookup table, because that could lead to impersonation. So there will not be an entity attribute, just a `learn-more-url` which publishers are responsible for if they want to add terms of service or similar.

The default is just showing the backend URL in the fashion of "your response will be sent to <this url>" in the dialog. If you want to show an entity name (eg: "your response will be sent to **Google** (backend.google.com)") then you need to be added to the lookup table.

**Comment by gmajoulet** [comment]
Got it, so to summarize:
- The button to open this dialog will automatically appear on all stories for all publishers
- They can add an extra "learn more" URL with the configuration you described
- They can add an extra "entity name" by sending a PR to our repository

**Comment by mszylkowski** [comment]
Yes! Also they can add a "learn more" url with the PR because sending the PR should provide all the necessary info to show the dialog.

**Comment by newmuis** [comment]
Wait -- the learn more URL here needs to be specified by the backend vendor, not the publisher

**Comment by mszylkowski** [comment]
In the case that backend vendors have not "claimed" that URL, shouldn't we allow publishers to specify the learn more URL in the component? I'd be ok with this

**Comment by newmuis** [comment]
My preference would be not to allow publishers to specify this, to:

1. Keep the API surface small(er) in a case where there's not clear demand, and
2. Avoid the potential (rare?) abuse vector that e.g. someone has an unclaimed backend and a publisher provides abusive "learn more" content

Let's allow backends to fully own this space; it's not much that they get.  The publisher has the entire story to place their links.
